File: frontend/lib/quiz-title.ts
const UNNAMED_PREFIX = "未命名题库";

export function isUnnamedTitle(title: string): boolean {
  const text = title.trim();
  return text === "" || text === UNNAMED_PREFIX;
}

export function nextUnnamedTitle(existing: string[]): string {
  const used = new Set(existing);
  let n = 1;
  while (used.has(`${UNNAMED_PREFIX}${n}`)) n += 1;
  return `${UNNAMED_PREFIX}${n}`;
}

export function withDuplicateSuffix(desired: string, existing: string[]): string {
  const text = desired.trim();
  if (!existing.includes(text)) return text;
  let n = 1;
  while (existing.includes(`${text}(${n})`)) n += 1;
  return `${text}(${n})`;
}

File: backend/app/services/quiz_title.py
UNNAMED_PREFIX = "未命名题库"


def is_unnamed_title(title: str) -> bool:
    text = (title or "").strip()
    return text == "" or text == UNNAMED_PREFIX


def next_unnamed_title(existing: list[str]) -> str:
    used = set(existing)
    n = 1
    while f"{UNNAMED_PREFIX}{n}" in used:
        n += 1
    return f"{UNNAMED_PREFIX}{n}"


def with_duplicate_suffix(desired: str, existing: list[str]) -> str:
    text = desired.strip()
    if text not in existing:
        return text
    n = 1
    while f"{text}({n})" in existing:
        n += 1
    return f"{text}({n})"


def uniquify_title(desired: str, existing: list[str]) -> str:
    text = (desired or "").strip()
    if is_unnamed_title(text):
        return next_unnamed_title(existing)
    return with_duplicate_suffix(text, existing)
